ui/components/app/wallet-overview/eth-overview.js

```diff
@@ -287,10 +302,19 @@ const EthOverview = ({ className, showAddress }) => {
                 history.push(SEND_ROUTE);
               });
             }}
+            tooltipRender={
+              signingEnabled
+                ? null
+                : (contents) => (
+                    <Tooltip title={t('accountCannotSign')} position="bottom">
+                      {contents}
+                    </Tooltip>
+                  )
+            }
```

Comment:
How do you see this code evolving? For example, it can start to grow fast if we need to handle 3 different tooltips or even more depending on the requirements.

Reply:
Excellent point, I refactored the way the tool tip is rendering to handle multiple conditions and messages for each button in this commit [1c26584](https://github.com/MetaMask/metamask-extension/pull/23178/commits/1c26584acef98c08c749e5f7b42bbcb70dea091e)
Please let me know what you think

Reply:
Looks great!